shared/query_engine/expressions/dataset.py

```diff
@@ -18,6 +20,13 @@ def build_expression_subquery(self, session) -> Selectable:
         sql_column = query_op.column
         sql_compare_operator = self.operator.operator_value
 
+        can_view = WorkingDir.can_member_view_datasets(session = session, project = self.project, dataset_ids = raw_scalar_value, member = self.member)
+        
+        if not can_view:
+            # TODO: How to handle this better? Should we throw? 
+            # # Also seems like an awkard place to do validation in general, is there a better place? Difficulty is that we need access to the dataset ids to do this validation.
+            self.log['error']['unauthorized'] = f'You do not have access to these datasets'
```

Comment:
Not sure the best way to handle the error case here. Also feels awkward to add validation here, but we need access to the dataset IDs to do the validation in the first place

I've been a bit confused about error handling in general across the codebase. Seen a mix of raising errors, using loggers, etc. Would be good to do an overview call for error handling